config/config.go

```diff
@@ -1,58 +1,60 @@
 package config
 
 import (
-	"fmt"
-	"os"
-	"path/filepath"
+	"path"
 	"reflect"
 	"time"
 
-	"github.com/BurntSushi/toml"
-	"github.com/ardanlabs/conf"
 	"github.com/pkg/errors"
-	"github.com/sirupsen/logrus"
+	"github.com/suutaku/nice-config/pkg/conf"
 )
 
 const (
-	DefaultConfigPath = "config/config.toml"
-	ConfigFileName    = "config.toml"
+	defaultConfigPath = "config"
+	configFileName    = "config.toml"
 	ServiceName       = "ssi-service"
 	ConfigExtension   = ".toml"
 
 	DefaultServiceEndpoint = "http://localhost:8080"
 )
 
+type Info struct {
+	Title string `toml:"title"`
+	Svn   string `toml:"svn"`
+	Desc  string `toml:"desc"`
+}
+
 type SSIServiceConfig struct {
-	conf.Version
-	Server   ServerConfig   `toml:"server"`
+	Info
+	Server   ServerConfig   `toml:"server" comment:"http service configuration"`
 	Services ServicesConfig `toml:"services"`
 }
 
 // ServerConfig represents configurable properties for the HTTP server
 type ServerConfig struct {
-	APIHost             string        `toml:"api_host" conf:"default:0.0.0.0:3000"`
-	DebugHost           string        `toml:"debug_host" conf:"default:0.0.0.0:4000"`
-	JagerHost           string        `toml:"jager_host" conf:"http://jaeger:14268/api/traces"`
-	JagerEnabled        bool          `toml:"jager_enabled" conf:"default:false"`
-	ReadTimeout         time.Duration `toml:"read_timeout" conf:"default:5s"`
-	WriteTimeout        time.Duration `toml:"write_timeout" conf:"default:5s"`
-	ShutdownTimeout     time.Duration `toml:"shutdown_timeout" conf:"default:5s"`
-	LogLocation         string        `toml:"log_location" conf:"default:log"`
-	LogLevel            string        `toml:"log_level" conf:"default:debug"`
-	EnableSchemaCaching bool          `toml:"enable_schema_caching" conf:"default:true"`
+	APIHost             string        `toml:"api_host"`
+	DebugHost           string        `toml:"debug_host"`
+	JagerHost           string        `toml:"jager_host"`
+	JagerEnabled        bool          `toml:"jager_enabled"`
+	ReadTimeout         time.Duration `toml:"read_timeout" comment:"5 seconds, time is in nanoseconds"`
```

Comment:
why remove the defaults, I believe they're useful!

Reply:
I never removed `default` values, just removed the `default` toml tags.  I have search to find a toml Marshaller which supported comments marshaling and it is `github.com/pelletier/go-toml/v2`.  as it's [README.md](https://github.com/pelletier/go-toml#support-for-default-struct-tag-has-been-dropped) described,**The recommended replacement is pre-filling the struct before unmarshaling.** ,so i modify code as recommended.